src/metabase/driver/generic_sql/util/unprepare.clj

```diff
@@ -14,11 +14,18 @@
   String  (unprepare-arg [this {:keys [quote-escape]}] (str \' (str/replace this "'" (str quote-escape "'")) \')) ; escape single-quotes
   Boolean (unprepare-arg [this _] (if this "TRUE" "FALSE"))
   Number  (unprepare-arg [this _] (str this))
-  Date    (unprepare-arg [this {:keys [iso-8601-fn]}] (first (hsql/format (hsql/call iso-8601-fn (hx/literal (u/date->iso-8601 this)))))))
+  UUID    (unprepare-arg [this _] (str \' this \'))
+  Date    (unprepare-arg [this {:keys [iso-8601-fn]}]
+                            (let [date-exp (u/date->iso-8601 this)]
+                              (first (hsql/format
+                                      (cond
+                                        (fn? iso-8601-fn) (iso-8601-fn date-exp)
+                                        (keyword? iso-8601-fn) (hsql/call iso-8601-fn (hx/literal date-exp))
+                                        :otherwise (hx/literal date-exp)))))))
```

Comment:
With `cond` the idiomatic way to do this is `:else`, not `:otherwise`

https://github.com/bbatsov/clojure-style-guide#else-keyword-in-cond